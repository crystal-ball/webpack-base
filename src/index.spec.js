const webpackConfigs = require('./index')

// Mocks to ensure snapshots output consistent values based on configs used
jest.mock('mini-css-extract-plugin')

describe('webpack-configs', () => {
  beforeEach(() => {
    // Ensure a consistent working directory is used for paths generated in
    // snapshots
    process.cwd = () => '/test/cwd/path'
  })

  test('returns expected dev configs', () => {
    // webpack-serve sets an environment variable
    process.env.WEBPACK_SERVE = true
    const baseConfigs = webpackConfigs({
      env: 'development',
      paths: { context: '/test' },
    })
    expect(baseConfigs).toMatchSnapshot()
  })

  test('returns expected prod configs', () => {
    // Remove the webpack serve env variable
    delete process.env.WEBPACK_SERVE
    const baseConfigs = webpackConfigs({
      env: 'production',
      paths: { context: '/test' },
    })
    expect(baseConfigs).toMatchSnapshot()
  })
})
