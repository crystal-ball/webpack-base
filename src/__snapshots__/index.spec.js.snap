// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack-configs returns expected dev configs 1`] = `
Object {
  "context": "/test",
  "devServer": Object {
    "compress": true,
    "contentBase": "/test/public",
    "historyApiFallback": true,
    "hot": true,
    "https": false,
    "overlay": Object {
      "errors": true,
      "warnings": false,
    },
    "port": 3000,
    "quiet": true,
  },
  "devtool": "eval",
  "entry": "/test/src/index.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.md\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
          },
          Object {
            "loader": "@inspirescript/magic-markdown-loader",
          },
        ],
      },
      Object {
        "include": Array [
          "/test/src/media/icons",
        ],
        "test": /\\\\\\.svg\\$/,
        "use": Array [
          Object {
            "loader": "svg-symbol-sprite-loader",
          },
        ],
      },
      Object {
        "exclude": Array [
          "/test/src/media/icons",
        ],
        "test": /\\\\\\.\\(jpe\\?g\\|png\\|gif\\|svg\\)\\$/i,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "name": "static/media/[name].[hash:8].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.txt\\$/,
        "use": Array [
          Object {
            "loader": "raw-loader",
          },
        ],
      },
      Object {
        "include": Array [
          "/test/src",
        ],
        "test": /\\\\\\.jsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
          },
          Object {
            "loader": "eslint-loader",
          },
        ],
      },
      Object {
        "test": /\\\\\\.scss\\$/,
        "use": Array [
          Object {
            "loader": "style-loader",
          },
          Object {
            "loader": "css-loader",
            "options": Object {
              "localIdentName": "[name]-[local]--[hash:5]",
            },
          },
          Object {
            "loader": "sass-loader",
            "options": Object {
              "includePaths": Array [
                "/test/src/styles",
              ],
            },
          },
        ],
      },
    ],
  },
  "optimization": Object {
    "runtimeChunk": true,
    "splitChunks": Object {
      "chunks": "all",
      "name": true,
    },
  },
  "output": Object {
    "filename": "static/js/[name].js",
    "hashDigestLength": 12,
    "path": "/test/dist",
    "publicPath": "/",
  },
  "plugins": Array [
    ProgressBarPlugin {
      "options": Object {
        "callback": [Function],
        "clear": false,
        "format": "  Hacking time... [:bar] [32m[1m:percent[22m[39m (:elapsed seconds) :msg",
      },
    },
    EnvironmentPlugin {
      "defaultValues": Object {
        "DEBUG": false,
        "PUBLIC_PATH": "/",
      },
      "keys": Array [
        "DEBUG",
        "PUBLIC_PATH",
      ],
    },
    HtmlWebpackPlugin {
      "options": Object {
        "cache": true,
        "chunks": "all",
        "chunksSortMode": "auto",
        "compile": true,
        "excludeChunks": Array [],
        "favicon": "/test/public/favicon.ico",
        "filename": "index.html",
        "hash": false,
        "inject": true,
        "meta": Object {},
        "minify": false,
        "showErrors": true,
        "template": "/test/public/index.html",
        "templateParameters": [Function],
        "title": "Webpack App",
        "xhtml": false,
      },
    },
    SVGSymbolSpritePlugin {
      "options": Object {
        "filename": "static/media/icon-sprite.[contenthash].svg",
        "injectSpriteId": true,
      },
    },
    Object {
      "apply": [Function],
    },
    FriendlyErrorsWebpackPlugin {
      "compilationSuccessInfo": Object {
        "messages": Array [
          "  ðŸŽ‰  [1m[32mBINGO[39m[22m ðŸŽ‰",
          "  Application running at [4m[34mhttp://localhost:3000[39m[24m",
        ],
        "notes": Array [],
      },
      "formatters": Array [
        [Function],
        [Function],
        [Function],
      ],
      "onErrors": undefined,
      "shouldClearConsole": true,
      "transformers": Array [
        [Function],
        [Function],
        [Function],
      ],
    },
    HotModuleReplacementPlugin {
      "fullBuildTimeout": 200,
      "multiStep": undefined,
      "options": Object {},
      "requestTimeout": 10000,
    },
  ],
  "resolve": Object {
    "alias": Object {},
    "extensions": Array [
      ".js",
      ".jsx",
      ".json",
    ],
    "modules": Array [
      "/test/src",
      "node_modules",
    ],
    "plugins": Array [
      Object {
        "apply": [Function],
      },
    ],
  },
}
`;

exports[`webpack-configs returns expected prod configs 1`] = `
Object {
  "bail": true,
  "context": "/test",
  "devtool": "source-map",
  "entry": "/test/src/index.js",
  "mode": "production",
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.md\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
          },
          Object {
            "loader": "@inspirescript/magic-markdown-loader",
          },
        ],
      },
      Object {
        "include": Array [
          "/test/src/media/icons",
        ],
        "test": /\\\\\\.svg\\$/,
        "use": Array [
          Object {
            "loader": "svg-symbol-sprite-loader",
          },
        ],
      },
      Object {
        "exclude": Array [
          "/test/src/media/icons",
        ],
        "test": /\\\\\\.\\(jpe\\?g\\|png\\|gif\\|svg\\)\\$/i,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "name": "static/media/[name].[hash:8].[ext]",
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.txt\\$/,
        "use": Array [
          Object {
            "loader": "raw-loader",
          },
        ],
      },
      Object {
        "include": Array [
          "/test/src",
        ],
        "test": /\\\\\\.jsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
          },
        ],
      },
      Object {
        "test": /\\\\\\.scss\\$/,
        "use": Array [
          "/test/path/to/mini-css-extract-loader",
          Object {
            "loader": "css-loader",
            "options": Object {
              "localIdentName": "[name]-[local]--[hash:5]",
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "plugins": Array [
                [Function],
              ],
            },
          },
          Object {
            "loader": "sass-loader",
            "options": Object {
              "includePaths": Array [
                "/test/src/styles",
              ],
            },
          },
        ],
      },
    ],
  },
  "optimization": Object {
    "runtimeChunk": true,
    "splitChunks": Object {
      "chunks": "all",
      "name": true,
    },
  },
  "output": Object {
    "filename": "static/js/[name].[chunkhash].js",
    "hashDigestLength": 12,
    "path": "/test/dist",
    "publicPath": "/",
  },
  "performance": Object {
    "hints": "warning",
    "maxAssetSize": 450000,
    "maxEntrypointSize": 1500000,
  },
  "plugins": Array [
    ProgressBarPlugin {
      "options": Object {
        "callback": [Function],
        "clear": false,
        "format": "  Hacking time... [:bar] [32m[1m:percent[22m[39m (:elapsed seconds) :msg",
      },
    },
    EnvironmentPlugin {
      "defaultValues": Object {
        "DEBUG": false,
        "PUBLIC_PATH": "/",
      },
      "keys": Array [
        "DEBUG",
        "PUBLIC_PATH",
      ],
    },
    HtmlWebpackPlugin {
      "options": Object {
        "cache": true,
        "chunks": "all",
        "chunksSortMode": "auto",
        "compile": true,
        "excludeChunks": Array [],
        "favicon": "/test/public/favicon.ico",
        "filename": "index.html",
        "hash": false,
        "inject": true,
        "meta": Object {},
        "minify": false,
        "showErrors": true,
        "template": "/test/public/index.html",
        "templateParameters": [Function],
        "title": "Webpack App",
        "xhtml": false,
      },
    },
    SVGSymbolSpritePlugin {
      "options": Object {
        "filename": "static/media/icon-sprite.[contenthash].svg",
        "injectSpriteId": true,
      },
    },
    Object {
      "apply": [Function],
    },
    NamedModulesPlugin {
      "options": Object {},
    },
    MonitorStats {
      "options": Object {
        "capture": true,
        "jsonOpts": Object {
          "chunkModules": true,
          "source": false,
        },
        "launch": undefined,
        "port": 8081,
        "target": "../monitor/stats.json",
      },
    },
    CleanWebpackPlugin {
      "options": Object {
        "allowExternal": false,
        "dry": false,
        "root": "/test/cwd/path",
        "verbose": false,
      },
      "paths": Array [
        "/test/dist",
      ],
    },
    DuplicatePackageCheckerPlugin {
      "options": Object {
        "emitError": true,
        "exclude": null,
        "showHelp": true,
        "strict": true,
        "verbose": true,
      },
    },
    MiniCSSExtractPlugin {
      "options": Object {
        "chunkFilename": "static/css/[name].[chunkhash].css",
        "filename": "static/css/[name].[chunkhash].css",
      },
    },
  ],
  "resolve": Object {
    "alias": Object {},
    "extensions": Array [
      ".js",
      ".jsx",
      ".json",
    ],
    "modules": Array [
      "/test/src",
      "node_modules",
    ],
    "plugins": Array [
      Object {
        "apply": [Function],
      },
    ],
  },
  "stats": Object {
    "excludeAssets": [Function],
    "modules": false,
  },
}
`;
